### ISSUE TITLE
numpy.squeeze compatibility fallback is wrong

### ISSUE THREAD

**Issue by njsmith** [numpy/numpy#10779]
`numpy.squeeze` used to take no arguments, but at some point gained an `axis` argument.

`numpy.squeeze(arr)` forwards to `arr.squeeze`.

`numpy.squeeze` therefore wants to handle both newer objects that support `arr.squeeze(axis=...)`, and also older objects that don't know about the axis argument.

Currently, the way it does this is:

```python
    try:
        # First try to use the new axis= parameter
        return squeeze(axis=axis)
    except TypeError:
        # For backwards compatibility
        return squeeze()
```

This is completely wrong. It should be:

```python
    if axis is None:  # the default, and only thing supported by older classes
        return squeeze()
    else:
        return squeeze(axis=axis)
```

**Comment by durgeshra** [comment]
HI! I will do it!

**Comment by hameerabbasi** [comment]
This was fixed in #10826 and should be closed.

**Comment by charris** [comment]
@hameerabbasi Thanks for the heads up.
